### PR TITLE
[RDAPI-22653] Changes for common criteria encryption for YAMLES

### PR DESCRIPTION
**Pull request by mcollins-axway**
Changes for private key and cert management.

Preview links 

https://deploy-preview-1544--axway-open-docs.netlify.app/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption/
https://deploy-preview-1544--axway-open-docs.netlify.app/docs/apim_yamles/yamles_edit/

### CODE REVIEW

File: content/en/docs/apim_yamles/yamles_edit.md [hunk 1]
```diff
@@ -61,58 +61,20 @@ Note that certificates and private keys that exist in the XML federated configur
...
+4. Create the `/Environment Configuration/Certificate Store/Axway.yaml` file in your YAML entity store:
 
     ```yaml
     ---
     type: Certificate
     fields:
       dname: Axway
-      key#refbinary: Axway-key.der
-      content#ref: Axway-cert.pem
+      key: '{{file "Axway-key.pem"}}'
```
Comment (dominiquemongelli) [inline content/en/docs/apim_yamles/yamles_edit.md:73]:
```suggestion
      key: '{{file "Axway-key.pem" pem}}'
```

File: content/en/docs/apim_yamles/yamles_edit.md [hunk 2]
```diff
@@ -61,58 +61,20 @@ Note that certificates and private keys that exist in the XML federated configur
...
+      content: '{{file "Axway-cert.pem"}}'
```
Comment (dominiquemongelli) [inline content/en/docs/apim_yamles/yamles_edit.md:74]:
```suggestion
      content: '{{file "Axway-cert.pem" pem}}'
```

File: content/en/docs/apim_yamles/yamles_edit.md (hunk 1 above)
Comment (dominiquemongelli) [reply content/en/docs/apim_yamles/yamles_edit.md:73]: "pem" options is mandatory if a pem with header and footer is provided.

File: content/en/docs/apim_yamles/yamles_edit.md [hunk 3]
```diff
@@ -148,17 +110,19 @@ Note that certificates and private keys that exist in the XML federated configur
...
     ```
 
-If the YAML configuration you are adding the certificate and private key into is encrypted with a passphrase, you will need to encrypt the private key file. To encrypt the private key file `Axway-key.der`, execute all the previous steps and complete the procedure with the following:
+If the YAML configuration you are adding the certificate and private key into is encrypted with a passphrase, you will need to encrypt the private key file. To encrypt the private key file `Axway-key.pem`, execute all the previous steps and do the following:
 
 ```shell
 cd apigateway/posix/bin
-./yamles encrypt --file ~/yaml/Environment\ Configuration/Certificate\ Store/Axway-key.der --passphrase changeme
+./yamles encrypt --file ~/yaml/Environment\ Configuration/Certificate\ Store/Axway-key.pem --source /home/user/yaml --passphrase changeme
 The file `/home/user/yaml/Environment Configuration/Certificate Store/Axway-key.der` has been encrypted
```
Comment (dominiquemongelli) [inline content/en/docs/apim_yamles/yamles_edit.md:118]:
```suggestion
The file `/home/user/yaml/Environment Configuration/Certificate Store/Axway-key.pem` has been encrypted
```

**Comment by mcollins-axway** [comment]
Andrea - this has been reviewed by another team member now so we are done from our side. Thanks

File: content/en/docs/apim_yamles/yamles_edit.md [hunk 4]
```diff
@@ -34,7 +34,7 @@ This section covers the steps required to add a new certificate and private key
 
 Note that certificates and private keys that exist in the XML federated configuration, which then get converted to YAML format, will be formatted as required with no additional steps required. In some cases, you might only need to add a new certificate to the YAML configuration, in this case the steps for the private key can be ignored.
 
-1. You must have a certificate in a `PEM` file and its related private key in separate `DER` file, for example, `Axway-cert.pem` and `Axway-key.der`. The private key should be unencrypted.
+1. You must have a certificate in a `PEM` file and its related private key in separate `PEM` file, for example, `Axway-cert.pem` and `Axway-key.pem`. The private key must be unencrypted.
 
 2. For testing purposes, you can generate an unencrypted private key and self-signed certificate PEM files using openssl as follows:
```
Comment (andreamussap) [inline content/en/docs/apim_yamles/yamles_edit.md:39]: @mcollins-axway  The instructions in the code block in this step aren't clear. Is this generated by the command, or did you/Team added this to help the user who's reading the documentation?

```
You are about to be asked to enter information that will be incorporated
into your certificate request.
What you are about to enter is what is called a Distinguished Name or a DN.
There are quite a few fields but you can leave some blank
```

'You're about to be asked for information... what you're about to enter is a DN' --> not clear. All the information that will be asked is this DN?

'There are quite a few fields but you can leave some blank' --> Irrelevant, should be removed. If you don't specify which fields are required, how will the user know?

File: content/en/docs/apim_yamles/yamles_edit.md [hunk 5]
```diff
@@ -148,17 +110,19 @@ Note that certificates and private keys that exist in the XML federated configur
...
 
-After you complete on the YAML configuration, create a `.tar.gz` file and deploy it to your running API Gateway.
+The YAML configuration referenced in the `--source` parameter of the `encrypt` command, must be the destination YAML configuration in order to ensure it is encrypted correctly.
+
+After you complete editing the YAML configuration, create a `.tar.gz` file and deploy it to your running API Gateway. Your API Gateway group must use the same encryption passphrase.
 
-{{< alert title="Note">}}Encrypting a private key via `openssl` and adding it to the YAML configuration is not supported. You must do this before packaging the `.tar.gz` and deploying to a gateway group that has a matching passphrase.{{< /alert >}}
+{{< alert title="Note">}}Encrypting a private key via `openssl` and adding it to the YAML configuration is not supported.{{< /alert >}}
 
 ## Manually create an entity instance YAML file
```
Comment (andreamussap) [inline content/en/docs/apim_yamles/yamles_edit.md:127]: @mcollins-axway The last step in the bullet list in this section is confusing: "Execute the first steps again to get all required and optional fields for each entity type.", but what are the 'first steps'?

Comment (andreamussap) [reply content/en/docs/apim_yamles/yamles_edit.md:127]: Another question in this section:  "Despite what is in the model, some fields that are said to be mandatory (cardinality=1 and no default value) are not mandatory. **Double check with Policy Studio** when in doubt." --> Can you make 'Double check with Policy Studio' a bit more helpful? Maybe add the section in Polity studio where the user can find this information ??

File: content/en/docs/apim_yamles/yamles_edit.md [hunk 6]
```diff
@@ -34,9 +34,9 @@ This section covers the steps required to add a new certificate and private key
 
 Note that certificates and private keys that exist in the XML federated configuration, which then get converted to YAML format, will be formatted as required with no additional steps required. In some cases, you might only need to add a new certificate to the YAML configuration, in this case the steps for the private key can be ignored.
 
-1. You must have a certificate in a `PEM` file and its related private key in separate `DER` file, for example, `Axway-cert.pem` and `Axway-key.der`. The private key should be unencrypted.
+1. You must have a certificate in a `PEM` file and its related private key in a separate `PEM` file. For example, `Axway-cert.pem` and `Axway-key.pem`. The private key must be unencrypted.
```
Comment (bordigoni) [inline content/en/docs/apim_yamles/yamles_edit.md:37]: "The private key cannot be a encrypted with openssl as it not supported."
might less confusing in the current context.

File: content/en/docs/apim_yamles/yamles_edit.md [hunk 7]
```diff
@@ -169,12 +133,12 @@ To create an entity instance of `NetService` type:
 
 * Search for YAML file `NetService.yaml` within the `META-INF/types` directory.
 * To know what fields can be used, move up to the ancestor types:
-    * `NetService.yaml` is contained within `JavaProcess` directory. Check fields within YAML entity type file `JavaProcess.yaml`.
-    * `JavaProcess.yaml` is contained within `Process` directory. Check fields within YAML entity type file `Process.yaml`.
+    * `NetService.yaml` is contained within `JavaProcess` directory. Check the fields within YAML entity type file `JavaProcess.yaml`.
+    * `JavaProcess.yaml` is contained within `Process` directory. Check tje fields within YAML entity type file `Process.yaml`.
```
Comment (bordigoni) [inline content/en/docs/apim_yamles/yamles_edit.md:137]:
```suggestion
    * `JavaProcess.yaml` is contained within `Process` directory. Check the fields within YAML entity type file `Process.yaml`.
```

File: content/en/docs/apim_yamles/yamles_edit.md (hunk 4 above)
Comment (mcollins-axway) [reply content/en/docs/apim_yamles/yamles_edit.md]: It is standard openssl stuff.

File: content/en/docs/apim_yamles/yamles_edit.md (hunk 5 above)
Comment (mcollins-axway) [reply content/en/docs/apim_yamles/yamles_edit.md:127]: I have reworded the "first steps" bit in this original commit as also thought it was confusing.

File: content/en/docs/apim_yamles/yamles_edit.md (hunk 6 above)
Comment (mcollins-axway) [reply content/en/docs/apim_yamles/yamles_edit.md:37]: Done

File: content/en/docs/apim_yamles/yamles_edit.md (hunk 5 above)
Comment (mcollins-axway) [reply content/en/docs/apim_yamles/yamles_edit.md:127]: Have just added a reword on the Policy Studio sentence

**Review by andreamussap — APPROVED** [review]

**Pull request merged**